### PR TITLE
v2-alpha.1

### PR DESCRIPTION
**Pull request by sygao**
## Dependency bump
- Across-board package dependencies updated, Python 3.9 required 
- Updated base Docker container from [continuumio/miniconda](https://hub.docker.com/r/continuumio/miniconda3) 4.9.2 => latest
- Updated `virtool_cli.vfam.msa` testing files from MUSCLE v3 to [MUSCLE v5](https://drive5.com/muscle5/)
- Modernized Poetry dev dependency specification

## Organizational changes

New, more hierarchal module structure.

- New `virtool_cli.utils` utility module
- `virtool_cli/utils.py` => `virtool_cli.utils.legacy`
- New `virtool_cli.vfam` submodule for `virtool hmm` code, new test folder structure to match
- `virtool_cli.run:cli` => `virtool_cli.cli:entry`

## Logging improvements
- Switching from Console to structlog 

## Deprecations
- Marking `virtool_cli.utils.parse_alpha()` as deprecated

## Misc
 
- Added new pull test workflow that includes alpha

### CODE REVIEW

File: virtool_cli/cli.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+from virtool_cli.hmm import hmm
+
+
+class OrderCommands(click.Group):
+  """
+  Modifies Group to list commands in the order of definition, 
+  instead of alphanumeric order
+  """
+  def list_commands(self, ctx: click.Context) -> list[str]:
+    return list(self.commands)
```
Comment (igboyes) [inline virtool_cli/cli.py:13]: Why?

File: virtool_cli/vfam/__init__.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from virtool_cli.vfam.vfam import *
+from virtool_cli.vfam.annotation import *
+from virtool_cli.vfam.collapse import *
+from virtool_cli.vfam.console import *
+from virtool_cli.vfam.filter import *
+from virtool_cli.vfam.markov import *
+from virtool_cli.vfam.msa import *
+from virtool_cli.vfam.polyprotein import *
```
Comment (igboyes) [inline virtool_cli/vfam/__init__.py:8]: Import specific names if you are going to do this.

File: virtool_cli/vfam/annotation.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 from pathlib import Path
 from typing import Tuple, List
 from Bio import SeqIO
-from virtool_cli.vfam_console import console
+from .console import console
```
Comment (igboyes) [inline virtool_cli/vfam/annotation.py:12]: Please use absolute imports only.

File: virtool_cli/utils/ref.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    """
+    return [otu for otu in src_path.iterdir() if otu.is_dir()]
+
+def generate_otu_dirname(otu_name: str):
+    """
+    """
+    no_whitespace = otu_name.replace(" ", "_")
+    no_plus = no_whitespace.replace('+', 'and')
+    no_symbols = re.split(r'[()/-]+', no_plus)
+    # otu_name.replace(" ", "_").replace("/", "_").lower()
```
Comment (igboyes) [inline virtool_cli/utils/ref.py:21]: Delete.

File: virtool_cli/utils/files.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+from pathlib import Path
+
+def rm_r(path: Path):
```
Comment (igboyes) [inline virtool_cli/utils/files.py:3]: Not used anywhere.

File: tests/vfam/fixtures_vfam.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+@pytest.fixture()
+def get_msa(output, fasta_files):
+    fasta_files = filter_on_coverage(fasta_files)
+    fasta_files = filter_on_number(fasta_files, 2)
+    return batch_muscle_call(fasta_files)
+
+
+@pytest.fixture()
+def get_hmm(get_msa):
+    return batch_hmm_call(get_msa)
```
Comment (igboyes) [inline tests/vfam/fixtures_vfam.py:94]: Change verb-based fixture names (`get_hmm`) to noun-based like `fasta_files`.

File: virtool_cli/utils/__init__.py [hunk 7]
```diff
@@ -0,0 +1,3 @@
+from .files import *
+from .legacy import *
+from .ref import *
```
Comment (igboyes) [inline virtool_cli/utils/__init__.py:3]: Import specific names or get rid of these lines.

**Review by igboyes — COMMENTED** [review]
* Suffix in some module names seems unnecessary (eg. `vfam/fixtures_vfam.py`).
* Please no wildcard or relative imports.

Looks good to me. Fix the issues and I will merge ASAP.

File: virtool_cli/cli.py (hunk 1 above)
Comment (sygao) [reply virtool_cli/cli.py:13]: It puts `virtool ref` above `virtool hmm` in the `virtool --help` command list.

```
❯ virtool --help
Usage: virtool [OPTIONS] COMMAND [ARGS]...

  ex. virtool ref --help

  note: virtool hmm is has additional bioconda dependencies, see documentation
  for installation instructions

Options:
  --help  Show this message and exit.

Commands:
  ref  Commands related to reference files.
  hmm  Commands related to Hidden Markov Models.
```

Obviously not necessary, but I thought I'd give it a try. Feel completely free to nix this if you'd like.

File: virtool_cli/utils/files.py (hunk 5 above)
Comment (sygao) [reply virtool_cli/utils/files.py:3]: Used in next alpha version, remove right now y/n?

Comment (igboyes) [reply virtool_cli/utils/files.py:3]: In that case, no. Please add docstring and improve name.

File: virtool_cli/cli.py (hunk 1 above)
Comment (igboyes) [reply virtool_cli/cli.py:13]: Kudos for exploring the nether reaches of `click`. I'd prefer to stick to the most basic usage as much as possible.

File: virtool_cli/utils/files.py (hunk 5 above)
Comment (sygao) [reply virtool_cli/utils/files.py]: Still working out the feature that needs this, so I'll remove from this version anyway.

File: .github/workflows/test.yml [hunk 8]
```diff
@@ -0,0 +1,33 @@
+name: ci
```
[the remark this answers is not in the document]
Comment (sygao) [reply .github/workflows/test.yml]: Correct. Added alpha branch to ci.yml instead. Sounds like that's a better idea?

File: virtool_cli/utils/ref.py (hunk 4 above)
Comment (sygao) [reply virtool_cli/utils/ref.py]: Removed line 21. Hopefully that is what was meant?

File: tests/vfam/fixtures_vfam.py (hunk 6 above)
Comment (sygao) [reply tests/vfam/fixtures_vfam.py]: Might want to double check the new names, I'm not sure if they're actually good names... there were some variable name conflicts when I tried the obvious options (hmm_files, msa_files)

**Comment by sygao** [comment]
Addressed everything I could. I'm a little iffy about the names on `tests/vfam/fixtures.py`. If you can think of better ones off the top of your head you can suggest them and I'll replace the ones in there now.

**Pull request merged**